packages/web/modals/review-order.tsx

```diff
@@ -649,9 +748,21 @@ export function ReviewOrder({
               <div className="flex w-full justify-between gap-3 pt-3">
                 <Button
                   mode="primary"
-                  onClick={confirmAction}
-                  disabled={isConfirmationDisabled}
+                  disabled={isConfirmationDisabled || isSpendingLimitExceeded}
                   className="body2 sm:caption !rounded-2xl"
+                  isLoading={oneClickTradingRef.current?.isLoading}
+                  onClick={async () => {
+                    if (
+                      !isOneClickTradingEnabled &&
+                      transaction1CTParams?.isOneClickEnabled &&
+                      oneClickTradingRef.current
+                    ) {
+                      await oneClickTradingRef.current.onStartTrading();
+                      setShouldTriggerTx(true);
+                    } else {
+                      confirmAction();
+                    }
+                  }}
```

Comment:
> Using refs to trigger a transaction is not recommended because it can lead to race conditions and produce unexpected results. Additionally, isLoading won’t indicate a loading state unless the component re-renders, which may not happen unless an unrelated value changes.

I have to disagree with that statement.
It shouldn't lead to race conditions since we create a reference to the function which is async and we can wait for the function to finish.
Regarding the isLoading it should work even without re-render, it is a reference and actually works quite well.
You could notice this because when accepting the transaction in review-modal enabling 1ct shows a spinner from onStartTrading.
That button before didn't have a spinner/loader and now it show one. Ref works pretty well and one of the benefits is that we could have that logic encapsulated inside a component instead of having in the parent in this case review-modal.

> I’m encountering an issue where I’m unable to initiate a swap immediately after a session is created because the 1CT session is still being set. When we attempt to start a 1CT transaction during this time, it results in an error because the session isn’t available yet.

That's not true it tries to submit the tx when it already have the session, after submitting the tx we change shouldTriggerTx state to true, but it only will only trigger when there is session. Here: https://github.com/osmosis-labs/osmosis-frontend/pull/3900/files#diff-25e66816c2ccd502684bb6420546938081ed0604faeb5dce40ad2277e8d237efR211

The problem is a different one related with this task is in the review modal which is something to be tackle in a different task thought.

Reply:
I understand your perspective, and I appreciate your detailed explanation. However, I’m still encountering an issue where I’m unable to initiate a swap immediately after a session is created because the 1CT session is still being set.